bentoml/_internal/server/grpc_app.py

```diff
@@ -52,7 +55,49 @@ def on_startup(self) -> OnStartup:
             for runner in self.bento_service.runners:
                 on_startup.append(runner.init_client)
 
-        return on_startup
+        async def wait_for_runner_ready():
+            ready_status = False
+            while not ready_status:
+                ready_status = all(
+                    await asyncio.gather(
+                        *(
+                            runner.runner_handle_is_ready()
+                            for runner in self.bento_service.runners
+                        )
+                    )
+                )
+
+        import yaml
+
+        start_time = time.time()
+        print("Waiting for runners %s to be ready.." % self.bento_service.runners)
+        with open(
+            "bentoml/_internal/configuration/default_configuration.yaml", "r"
+        ) as f:
+            timeout = yaml.load(f)["runners"]["timeout"]
```

Comment:
get timeout via `BentoMLContainer.runners.timeout`
```python
from simple_di import Provide, inject

...
@inject
def on_startup(self, timeout: int = Provide[BentoMLContainer.runners.timeout]):
...
```

Also on_startup should be sync not async.